src/stakers/stakers.service.ts

```diff
@@ -44,4 +44,47 @@ export class StakersService {
     this.logger.log('Staker data fetched', { withdraw_authority, result });
     return balances;
   }
+
+  async getAllNativeStakeBalances(
+    startDate: string,
+    endDate: string,
+  ): Promise<{ [key: string]: NativeStakeBalanceDto[] } | null> {
+    if (!startDate) {
+      startDate = new Date(0).toISOString();
+    }
+
+    if (!endDate) {
+      endDate = new Date(Date.now()).toISOString();
+    }
+
+    const result = await this.rdsService.pool.any(sql.unsafe`
+        SELECT *
+        FROM native_stake_accounts
+        LEFT JOIN snapshots ON snapshots.snapshot_id = native_stake_accounts.snapshot_id
+        WHERE snapshots.created_at >= ${startDate} AND snapshots.created_at <= ${endDate}
+      `);
+
+    if (!result || result.length === 0) {
+      this.logger.warn('Stakes not found!');
+      return null;
+    }
+
+    const ownerBalances: { [key: string]: NativeStakeBalanceDto[] } = {};
```

Comment:
Same case like in the comment on the function signature above, using `NativeStakeBalancesDto` should be ok here.